### PR TITLE
chore: Misc. component cleanup

### PR DESCRIPTION
**Pull request by jadebenn**
Formatting and syntax cleanup plus making eReplicaComponent values constexpr

### CODE REVIEW

**Comment by EmosewaMC** [comment]
is inline static const not equivalent to constexpr?  it's just a number at the end of the day, so this shouldnt make a difference?

File: dGame/dComponents/SoundTriggerComponent.h [hunk 1]
```diff
@@ -58,19 +59,19 @@ struct MixerProgram{
...
-	std::vector<MusicCue> m_MusicCues = {};
-	std::vector<MusicParameter> m_MusicParameters = {};
-	std::vector<GUIDResults> m_2DAmbientSounds = {};
-	std::vector<GUIDResults> m_3DAmbientSounds = {};
-	std::vector<MixerProgram> m_MixerPrograms = {};
+	std::vector<MusicCue> m_MusicCues{};
+	std::vector<MusicParameter> m_MusicParameters{};
+	std::vector<GUIDResults> m_2DAmbientSounds{};
+	std::vector<GUIDResults> m_3DAmbientSounds{};
+	std::vector<MixerProgram> m_MixerPrograms{};
```
Comment (EmosewaMC) [inline dGame/dComponents/SoundTriggerComponent.h:74]: nit: no brace needed here for a default constructor call anyways

**Review by EmosewaMC — DISMISSED** [review]

**Comment by jadebenn** [comment]
> is inline static const not equivalent to constexpr? it's just a number at the end of the day, so this shouldnt make a difference?

Difference is that constexpr values can be used in compile-time calculations. Having those enums available at compile-time might be needed for some functionality I want to implement, and even if it isn't, it certainly can't hurt.

**Comment by EmosewaMC** [comment]
> > is inline static const not equivalent to constexpr? it's just a number at the end of the day, so this shouldnt make a difference?
> 
> Difference is that static constexpr members can be used in compile-time calculations. Having those members available at compile-time might be needed for some functionality I want to implement, and even if it isn't, it certainly can't hurt.

I still dont believe it makes a difference for integral types like an integer but sure

**Review by EmosewaMC — APPROVED** [review]

**Comment by EmosewaMC** [comment]
image

**Comment by jadebenn** [comment]
Looks like I was tired enough to mistake the "commit message" field for the "directory" field

File: dGame/dComponents/ScriptedActivityComponent.h [hunk 2]
```diff
@@ -6,9 +6,9 @@
 
 class Entity;
 
-class ScriptedActivityComponent : public ActivityComponent {
+class ScriptedActivityComponent final : public ActivityComponent {
```
Comment (aronwk-aaron) [inline dGame/dComponents/ScriptedActivityComponent.h:9]: this will not be final in the future.

**Review by aronwk-aaron — APPROVED** [review]

**Review by EmosewaMC — APPROVED** [review]
pending CI.

**Pull request merged**